networking/gcp/psa/variables.tf

```
@@ -0,0 +1,62 @@
+######## SECTION - VPC to get
+
+variable "vpc_link" {
+  description = "The vpc on which the psa will be created"
+  type        = string
+}
+
+######## SECTION - PSA 
+
+variable "service_name" {
+  description = "The private service access to create"
+  type        = string
+  default     = "servicenetworking.googleapis.com"
+}
+
+######## SECTION - GLOBAL ADDRESS
+
+variable "global_address_name" {
+  description = "Name of the resource."
+  type        = string
+  validation {
+    condition     = can(regex("[a-z]([-a-z0-9]*[a-z0-9])?.", var.global_address_name))
+    error_message = "The name of the global adress must match the following regular expression : [a-z]([-a-z0-9]*[a-z0-9])?."
+  }
+}
+
+variable "global_address_description" {
+  description = "An optional description for the global_address resource"
+  type        = string
+  default     = null
+}
+
+variable "global_address_ip" {
+  description = "The static IP represented by this resources."
+  type        = string
+  default     = null
+}
+
+variable "global_address_prefix_length" {
+  description = "The prefix length if the resource represents an IP range."
+  type        = number
+  default     = 0
+}
```

Default should be 32 for IPv4 and 128 for IPv6. I think it handled directly by the `google_compute_global_address` resource if it is null.

```suggestion
variable "prefix_length" {
  description = "The prefix length if the resource represents an IP range."
  type        = number
  default     = null
}
```